rosbag2_converter_default_plugins/src/rosbag2_converter_default_plugins/cdr/cdr_converter.cpp

```diff
@@ -63,37 +63,50 @@ std::string get_package_library_path(const std::string & package_name)
 CdrConverter::CdrConverter()
 {
   auto library_path = get_package_library_path("rmw_fastrtps_cpp");
-  std::shared_ptr<Poco::SharedLibrary> library;
-  try {
-    library = std::make_shared<Poco::SharedLibrary>(library_path);
-  } catch (Poco::LibraryLoadException &) {
-    throw std::runtime_error(
-            std::string("poco exception: library could not be found:") + library_path);
+
+  std::unique_ptr<rcutils_shared_library_t> library =
+    std::make_unique<rcutils_shared_library_t>();
+  *library = rcutils_get_zero_initialized_shared_library();
+
+  rcutils_ret_t ret = rcutils_load_shared_library(library.get(), library_path.c_str());
+  if (ret != RCUTILS_RET_OK) {
+    if (ret == RCUTILS_RET_BAD_ALLOC) {
+      throw std::runtime_error{"rcutils shared_library exception: failed to allocate memory"};
+    } else if (ret == RCUTILS_RET_INVALID_ARGUMENT) {
+      throw std::runtime_error{"rcutils shared_library exception: invalid arguments"};
+    } else {
+      throw std::runtime_error{"rcutils shared_library exception: library could not be found:" +
+              library_path};
+    }
```

Comment:
There is `rclcpp::throw_from_rcl_error`, maybe we can leverage this? Even though strictly speaking, `rcutils_ret_t` and `rcl_ret_t` might become different datatypes at some point.

```c++
RCLCPP_PUBLIC
void
throw_from_rcl_error [[noreturn]] (
  rcl_ret_t ret,
  const std::string & prefix = "",
  const rcl_error_state_t * error_state = nullptr,
  void (* reset_error)() = rcl_reset_error);
```